tests/unit-tests/plugin-manager.php

```diff
@@ -0,0 +1,130 @@
+<?php
+/**
+ * Tests the plugin management functionality.
+ *
+ * @package Newspack\Tests
+ */
+
+use Newspack\Plugin_Manager;
+
+/**
+ * Test plugin management functionality.
+ */
+class Newspack_Test_Plugin_Manager extends WP_UnitTestCase {
+
+	/**
+	 * Plugin slug/folder.
+	 *
+	 * @var string
+	 */
+	protected $plugin_slug = 'hello-dolly';
+
+	/**
+	 * Plugin file path.
+	 *
+	 * @var string
+	 */
+	protected $plugin_file = 'hello-dolly/hello.php';
+
+	/**
+	 * URL to plugin download.
+	 *
+	 * @var string
+	 */
+	protected $plugin_url = 'https://downloads.wordpress.org/plugin/hello-dolly.1.6.zip';
+
+	/**
+	 * Compatibility checks and clean up.
+	 */
+	public function setUp() {
+		// These tests can't run on environments where we can't install plugins e.g. VIP Go.
+		if ( ! Plugin_Manager::can_install_plugins() ) {
+			$this->markTestSkipped( 'Plugin installation is not allowed in the environment' );
+		}
+
+		// Clean up any lingering plugin installs that may exist from a previous test.
+		Plugin_Manager::uninstall( $this->plugin_file );
+	}
+
+	/**
+	 * Test the plugin-slug-from-URL parser.
+	 */
+	public function test_get_plugin_slug_from_url() {
+		$this->assertEquals( 'hello-dolly', Plugin_Manager::get_plugin_slug_from_url( 'https://downloads.wordpress.org/plugin/hello-dolly.1.6.zip' ) );
+		$this->assertEquals( 'hello-dolly', Plugin_Manager::get_plugin_slug_from_url( 'https://downloads.wordpress.org/plugin/hello-dolly.zip/' ) );
+		$this->assertEquals( 'hello-dolly', Plugin_Manager::get_plugin_slug_from_url( 'https://downloads.wordpress.org/plugin/hello-dolly.zip?foo=blah&1' ) );
+		$this->assertEquals( false, Plugin_Manager::get_plugin_slug_from_url( 'https://downloads.wordpress.org/plugin/hello-dolly' ) );
+		$this->assertEquals( false, Plugin_Manager::get_plugin_slug_from_url( 'https://downloads.wordpress.org/plugin/hello-dolly.tar' ) );
+		$this->assertEquals( false, Plugin_Manager::get_plugin_slug_from_url( 'hello-dolly' ) );
```

Comment:
Should we add some more extreme inputs here? For example, what if we passed an;

* `''` (empty string),
* `bool`,
* `'WP_Error',
* `int`